subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -228,11 +231,25 @@ private TransientConfigurationResults loadTransientGraphResults(SelectedArtifact
     /**
      * Recursive but excludes unsuccessfully resolved artifacts.
      */
-    public Set<File> getFiles(Spec<? super Dependency> dependencySpec) {
-        Set<File> files = Sets.newLinkedHashSet();
-        FilesAndArtifactCollectingVisitor visitor = new FilesAndArtifactCollectingVisitor(files);
-        visitArtifacts(dependencySpec, ImmutableAttributes.EMPTY, selectedArtifacts, selectedFileDependencies, visitor);
-        files.addAll(getFiles(filterUnresolved(visitor.artifacts)));
+    public Set<File> getFiles(final Spec<? super Dependency> dependencySpec) {
+        final Set<ResolvedArtifactResult> artifactResults = Sets.newLinkedHashSet();
+        final Set<File> files = Sets.newLinkedHashSet();
+        final ResolvedArtifactCollectingVisitor visitor = new ResolvedArtifactCollectingVisitor(artifactResults);
```

Comment:
This behaves differently WRT lenience. The `ResolvedArtifactCollectingVisitor` collects all failures for later throwing. Which means using it here that we're ignoring all failures. Instead, we only want to ignore `ArtifactResolveException` when `artifact.getFile()` is called.